### PR TITLE
ddl: reduce the time to run unit tests

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Reduce the time to run unit tests in `ddl` package.

### What is changed and how it works?

- Split tests. Relate to #8612.

- Remove some operations of "drop table" in tests.

```
Before this PR:
ok  	github.com/pingcap/tidb/ddl	25.074s	coverage: 79.5% of statements
ok  	github.com/pingcap/tidb/ddl/failtest	7.919s	coverage: 0.0% of statements
After this PR:
ok  	github.com/pingcap/tidb/ddl	17.210s	coverage: 79.5% of statements
ok  	github.com/pingcap/tidb/ddl/failtest	3.240s	coverage: 0.0% of statements
```
### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8585)

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -459,7 +462,12 @@ func (d *ddl) GetID() string {
 	return d.uuid
 }
 
-func checkJobMaxInterval(job *model.Job) time.Duration {
+func (d *ddl) checkJobMaxInterval(job *model.Job) time.Duration {
+	// testCheckJobMaxInterval's value may not be 0 only in tests.
+	if d.testCheckJobMaxInterval != 0 {
+		return d.testCheckJobMaxInterval
```
Comment (winkyao) [inline ddl/ddl.go]: It is dangerous here, change a testCheckJobMaxInterval maybe neglected by the reviewer. How about make the hard code time in line 474 and line 477 to a variable, then we can modify that variable.

Comment (zimulala) [reply ddl/ddl.go]: We don‘t need to care `testCheckJobMaxInterval`, it only can be set by `SetCheckJobMaxInterval`, and `SetCheckJobMaxInterval` is in ddl_test.

**Comment by zz-jason** [comment]
@zimulala seems CI is failed?

**Comment by zimulala** [comment]
@zz-jason 
Not caused by this PR.

**Comment by zimulala** [comment]
PTAL @zz-jason @winkyao

**Comment by winkyao** [comment]
This PR needn't 3 LGTM，it just change the test cases.

**Comment by zimulala** [comment]
PTAL @crazycs520 @ciscoxll

**Pull request merged**